### PR TITLE
[inductor] preserve scan carry strides with shape padding

### PR DESCRIPTION
**Pull request by ydwu4**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #183384
* #183383

Scan lowering decomposes scan into while_loop, whose carried values must preserve metadata across iterations. Shape padding can return same-sized matmul results with padded strides, so normalize next_carry back to the incoming carry stride and cover the forced-padding regression.

Authored with Codex.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

File: torch/_inductor/fx_passes/post_grad.py [hunk 1]
```diff
@@ -743,6 +743,23 @@ def body_fn(*flat_args):
...
+                        else torch.empty_strided(
+                            next_carry_elem.size(),
+                            carry_elem.stride(),
+                            device=next_carry_elem.device,
+                            dtype=next_carry_elem.dtype,
+                            layout=next_carry_elem.layout,
+                            requires_grad=next_carry_elem.requires_grad,
+                        ).copy_(next_carry_elem)
+                    )
+                    for next_carry_elem, carry_elem in zip(next_carry, carry)
```
Comment (eellison) [inline torch/_inductor/fx_passes/post_grad.py:751]: Instead, can we use the require_exact_strides from inductor ? or graph user_visible_outputs?

Comment (ydwu4) [reply torch/_inductor/fx_passes/post_grad.py:751]: I tried relying on while_loop’s existing ExternKernel.require_exact_strides path, but it happens too late for this case. The failure occurs while tracing the scan-to-while_loop replacement in post_grad passes: while_loop_fake_tensor_mode validates that carried inputs and body outputs have matching metadata before Inductor IR lowering runs. With shape padding, the body output fake stride is already different, so we never reach WhileLoop.create.  Similarly graph.user_visible_outputs would not fix the body-output/carry stride check in post_grad.

Added an inductor_prims.force_exact_stride gives fake tensor propagation the corrected stride metadata during tracing, and then lowers to ir.ExternKernel.require_exact_strides in Inductor. wdyt?
